prowler/providers/azure/services/monitor/monitor_service.py

```diff
@@ -0,0 +1,157 @@
+    #            )
+    #            for log in logs:
+    #                activity_logs[subscription].append(
+    #                    ActivityLog(
+    #                        id=log.id,
+    #                        resource_group=log.resource_group_name,
+    #                        resource_id=log.resource_id,
+    #                        operation_name=log.operation_name,
+    #                        status=log.status,
+    #                        event_timestamp=log.event_timestamp,
+    #                        caller=log.caller,
+    #                        correlation_id=log.correlation_id,
+    #                        category=log.category,
+    #                        level=log.level,
+    #                        event_data_id=log.event_data_id,
+    #                        sub_status=log.sub_status,
+    #                    )
+    #                )
+    #        except Exception as error:
+    #            logger.error(
+    #                f"Subscription name: {subscription} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+    #            )
+    #    #print(activity_logs)
+    #    print("HOLA")
+    #    return activity_logs
+
+    # def __get_activity_log_alerts__(self):
+    #    logger.info("Monitor - Getting activity log alerts...")
+    #    activity_log_alerts = {}
+    #    for subscription, client in self.clients.items():
+    #        try:
+    #            activity_log_alerts.update({subscription: []})
+    #            alerts = client.activity_log_alerts.list_by_subscription_id()
+    #            for alert in alerts:
+    #                print(alert.condition.all_of[0].equals[0].value)
+    #                activity_log_alerts[subscription].append(
+    #                    ActivityLogAlert(
+    #                        id=alert.id,
+    #                        name=alert.name,
+    #                        description=alert.description,
+    #                        enabled=alert.enabled,
+    #                        scopes=alert.scopes,
+    #                        condition=alert.condition,
+    #                        actions=alert.actions,
+    #                        last_updated_time=alert.last_updated_time,
+    #                        provisioning_state=alert.provisioning_state,
+    #                    )
+    #                )
+    #        except Exception as error:
+    #            logger.error(
+    #                f"Subscription name: {subscription} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+    #            )
+    #    return activity_log_alerts
+
+
+# @dataclass
+# class ActivityLog:
+#    id: str
+#    resource_group: str
+#    resource_id: str
+#    operation_name: str
+#    status: str
+#    event_timestamp: str
+#    caller: str
+#    correlation_id: str
+#    category: str
+#    level: str
+#    event_data_id: str
+#    sub_status: str
+
+
+@dataclass
```

Comment:
Why are you not using Pydantic here?

Reply:
Because of the types, logs is a list that consists of 2 strings, a boolean and a RetentionPolicy that consists of an integer and a boolean, when I structure this using Pydantic somehow Python doesn't get the data correctly and execution fails.

Reply:
Can you set the type for the `logs`? Like `list[WhateverIsNamedInAzureMonitor]`

Reply:
That is what I tried, but Python wasn't getting the types correctly. This is what I did:

class RetentionPolicy(BaseModel):
    enabled: bool
    days: int   

class LogSettings(BaseModel):
    category: str
    category_group: str
    enabled: bool
    retention_policy: RetentionPolicy

class DiagnosticSetting(BaseModel):
    id: str
    logs: List[LogSettings]

Reply:
That's correct, that was the old version trying to use Pydantic, the new one is:


from azure.mgmt.monitor.models import LogSettings
from pydantic import BaseModel

class DiagnosticSetting(BaseModel):
    id: str
    logs: LogSettings

Reply:
That's good, push that change then!